ComfortZoneApp/app/src/main/java/com/example/comfortzone/WeatherClient.java

```diff
@@ -86,22 +86,23 @@ public void onResponse(@NonNull Call call, @NonNull Response response) throws IO
         });
     }
 
-    public void getGroupWeatherUrl() {
+    public void getGroupWeatherUrl(GroupUrlCallback callback) {
         getCityData(new CityListCallback() {
             @Override
             public void cityList(List<City> cityList) {
-                List<List<String>> batchCityUrls = new ArrayList<>();
                 List<String> ids = cityList.stream().map(city -> String.valueOf(city.getId())).collect(Collectors.toList());
                 List<List<String>> batchesCities = Lists.partition(ids, MAX_API_CITIES);
-                for (List<String> batchCity : batchesCities) {
-                    String idString = String.join(",", batchCity);
-                    String url = String.format(API_GROUP_BASE_URL + "&id=%s", idString);
-                    batchCityUrls.add(Collections.singletonList(url));
-                }
+                List<String> batchCityUrls = batchesCities.stream().map(batch -> getBatchUrl(batch)).collect(Collectors.toList());
+                callback.weatherUrlGroupIds(batchCityUrls);
             }
         });
     }
 
+    private String getBatchUrl(List<String> batchCity) {
+        String idString = String.join(",", batchCity);
+        return String.format(API_GROUP_BASE_URL + "&id=%s", idString);
```

Comment:
I tried using addQueryParameter, but because the string of ids need to be separated by commas, I wasn't able to use addQueryParameter since they turn all commas into a string of characters